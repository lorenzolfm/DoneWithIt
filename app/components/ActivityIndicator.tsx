import React from 'react';
import LottieView from 'lottie-react-native';
import { StyleSheet, View } from 'react-native';

type Props = { visible: boolean }

export const ActivityIndicator = ({ visible = false }: Props) => {
  if (!visible) return null;

  return (
    <View style={styles.overlay}>
      <LottieView
        autoPlay
        loop
        source={require('../assets/animations/loader.json')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  overlay:  {
    position: 'absolute',
    backgroundColor: 'white',
    height: '100%',
    width: '100%',
    zIndex: 1,
    opacity: 0.8,
  }
})
