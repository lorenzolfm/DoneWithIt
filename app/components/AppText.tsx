import React from 'react';
import { Text } from 'react-native';

import defaultStyles from '../config/styles';

type AppTextProps = {
  children?: React.ReactNode,
  style?: object,
  [otherProp:string]: any,
};
export const AppText = ({ children, style, ...otherProps }: AppTextProps) => {
  return (
    <Text style={[defaultStyles.text, style]} {...otherProps}>{children}</Text>
  );
};
