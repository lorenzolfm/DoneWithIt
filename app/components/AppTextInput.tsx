import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import defaultStyles from '../config/styles';

type Props = {
  icon?: any,
  placeholder?: string,
  [otherProp:string]: any,
};
export const AppTextInput = ({ icon, placeholder, ...otherProps }: Props) => {
  return (
    <View style={styles.container}>
      {icon && <MaterialCommunityIcons name={icon} size={20} color={defaultStyles.colors.medium} style={styles.icon} />}
      <TextInput style={defaultStyles.text} placeholder={placeholder} {...otherProps}/>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: defaultStyles.colors.light,
    borderRadius: 25,
    flexDirection: 'row',
    width: '100%',
    padding: 15,
    marginVertical: 10,
    alignItems: 'center',
  },
  icon: {
    marginRight: 10,
  }
});
