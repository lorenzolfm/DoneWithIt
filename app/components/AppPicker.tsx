import React, { useState } from 'react';
import { StyleSheet, View, TouchableWithoutFeedback, Modal, Button, FlatList } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import defaultStyles from '../config/styles';
import { AppText } from './AppText';
import { Screen } from './Screen';

import { Category } from '../types';
import { PickerItem } from './PickerItem';

type Props = {
  icon?: any,
  placeholder?: string,
  items?: Category[],
  width?: number | string,
  selectedItem: Category,
  onSelectItem: Function,
};
export const AppPicker = ({ icon, placeholder, items, width = '100%', selectedItem, onSelectItem }: Props) => {
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <>
      <TouchableWithoutFeedback onPress={() => setModalVisible(true)}>
        <View style={[styles.container, { width }]}>
          {icon && <MaterialCommunityIcons name={icon} size={20} color={defaultStyles.colors.medium} style={styles.icon} />}
          {selectedItem ?
            <AppText style={styles.text}>{selectedItem.label}</AppText> :
            <AppText style={styles.placeholder}>{placeholder}</AppText>
          }
          <MaterialCommunityIcons
            name='chevron-down'
            size={20}
            color={defaultStyles.colors.medium}
          />
        </View>
      </TouchableWithoutFeedback>
      <Modal visible={modalVisible} animationType="slide">
        <Screen>
          <Button title="Close" onPress={() => setModalVisible(false)} />
          <FlatList
            data={items}
            keyExtractor={item => item.value.toString()}
            renderItem={
              ({ item }) =>
                <PickerItem label={item.label} onPress={() => { setModalVisible(false); onSelectItem(item) }} />
            }
          />
        </Screen>
      </Modal>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: defaultStyles.colors.light,
    borderRadius: 25,
    flexDirection: 'row',
    width: '100%',
    padding: 15,
    marginVertical: 10,
    alignItems: 'center',
  },
  icon: {
    marginRight: 10,
  },
  text: {
    flex: 1,
  },
  placeholder: {
    flex: 1,
    color: defaultStyles.colors.medium,
  }
});
