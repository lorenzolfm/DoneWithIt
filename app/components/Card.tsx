import React from 'react';
import { Image, StyleSheet, View } from "react-native";
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';

import defaultStyles from '../config/styles';
import { AppText } from "./AppText";

type CardProps = {
  title: string,
  subtitle: string,
  imageUrl: string,
  onPress: () => void,
}

export const Card = ({ title, subtitle, imageUrl, onPress }: CardProps) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.card}>
        <Image source={{uri: imageUrl}} style={styles.image}/>
        <View style={styles.detailsContainer}>
          <AppText style={styles.title} numberOfLines={1}>{title}</AppText>
          <AppText style={styles.subtitle} numberOfLines={2}>{subtitle}</AppText>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  card: {
    borderRadius: 15,
    backgroundColor: defaultStyles.colors.white,
    marginBottom: 20,
    overflow: 'hidden',
  },
  detailsContainer: { padding: 20 },
  image: {
    width: '100%',
    height: 200,
  },
  title: {
    marginBottom: 10,
  },
  subtitle: {
    color: defaultStyles.colors.secondary,
    fontWeight: 'bold',
  }
});
