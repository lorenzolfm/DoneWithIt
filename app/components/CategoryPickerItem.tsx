import React from 'react';
import { GestureResponderEvent, StyleSheet, View } from 'react-native';

import { AppText } from './AppText';
import { Icon } from './Icon';

type Props = {
  item: any
  onPress: (event: GestureResponderEvent) => void,
}

export const CategoryPickerItem = ({ item, onPress }: Props) => {
  return (
    <View style={styles.container}>
      <Icon backgroundColor={item.backgroundColor} name={item.icon} size={80} />
      <AppText style={styles.label}>{item.label}</AppText>
    </View>
  )
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    paddingVertical: 15,
    alignItems: 'center',
    width: '33%',
  },
  label: {
    marginTop: 5,
    textAlign: 'center',
  },
});
