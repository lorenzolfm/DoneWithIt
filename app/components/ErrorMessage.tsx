import React from 'react';
import { StyleSheet } from 'react-native';

import { AppText } from './AppText';

type Props = {
  error?: string,
  visible: boolean | undefined,
};

export const ErrorMessage = ({ error, visible }: Props) => {
  if (!visible || !error) return null;

  return (
    <AppText style={styles.error}>{error}</AppText>
  );
};

const styles = StyleSheet.create({
  error: { color: 'red' },
});
