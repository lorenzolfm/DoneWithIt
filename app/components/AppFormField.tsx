import React from 'react';
import { useFormikContext } from 'formik';

import { AppTextInput } from './AppTextInput';
import { ErrorMessage } from './ErrorMessage';

type Props = {
  name: string,
  [otherProp: string]: any,
};
export const AppFormField = ({ name, ...otherProps }: Props) => {
  const { setFieldTouched, handleChange, errors, touched } = useFormikContext();

  return (
    <>
      <AppTextInput
        onBlurEvent={() => setFieldTouched(name)}
        onChangeText={handleChange(name)}
        {...otherProps}
      />
      <ErrorMessage error={errors[name]} visible={touched[name]} />
    </>
  );
};
