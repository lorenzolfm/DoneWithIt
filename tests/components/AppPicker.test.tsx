import { shallow } from 'enzyme';
import { AppPicker } from '../../app/components/AppPicker';

describe('AppPicker', () => {
  it('snapshot matches', () => {
    expect(shallow(<AppPicker />)).toMatchSnapshot();
  });
});
