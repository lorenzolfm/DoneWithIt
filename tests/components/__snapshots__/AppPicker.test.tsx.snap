// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPicker snapshot matches 1`] = `
<Fragment>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f4f4",
            "borderRadius": 25,
            "flexDirection": "row",
            "marginVertical": 10,
            "padding": 15,
            "width": "100%",
          },
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <AppText
        style={
          Object {
            "color": "#6e6969",
            "flex": 1,
          }
        }
      />
      <View
        color="#6e6969"
        name="chevron-down"
        size={20}
      />
    </View>
  </TouchableWithoutFeedback>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <Screen>
      <Button
        onPress={[Function]}
        title="Close"
      />
      <FlatList
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </Screen>
  </Modal>
</Fragment>
`;
